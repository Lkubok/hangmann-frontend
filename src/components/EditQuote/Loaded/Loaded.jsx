import React, { Component } from "react";
import { connect } from "react-redux";
import QuoteForm from "../../QuoteForm/QuoteForm";
import * as selectors from "../../../reducers/selectors";
import PropTypes from "prop-types";
import "./Loaded.scss";

export class Loaded extends Component {
  render() {
    const { quote, quoteAuthor, insertAuthor, lang } = this.props.fetchedQuote;
    return (
      <div>
        <QuoteForm
          id={this.props.quoteId}
          touched=""
          error=""
          quote={quote}
          quoteAuthor={quoteAuthor}
          insertAuthor={insertAuthor}
          lang={lang}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  fetchedQuote: selectors.getFetchedQuote(state)
});

const mapDispatchToProps = dispatch => ({});

Loaded.propTypes = {
  fetchedQuote: PropTypes.object
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Loaded);
