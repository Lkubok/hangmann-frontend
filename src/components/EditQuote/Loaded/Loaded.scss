@import "../../../scss/mixins.scss";
@import "../../../scss/variables.scss";

.loaded-content {
  width: 50%;
  height: 600px;
}
