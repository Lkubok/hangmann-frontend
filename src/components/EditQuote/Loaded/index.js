import Loaded from "./Loaded";
export default Loaded;
