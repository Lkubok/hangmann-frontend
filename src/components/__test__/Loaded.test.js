import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Loaded } from "../EditQuote/Loaded/Loaded";
Enzyme.configure({ adapter: new Adapter() });

describe("Loaded component", () => {
  let props;
  beforeEach(() => {
    props = {
      fetchedQuote: {
        quote: "test",
        quoteAuthor: "test",
        insertAuthor: "test",
        lang: "test"
      }
    };
  });
  it("Should match snapshot", () => {
    const wrapper = shallow(<Loaded {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
