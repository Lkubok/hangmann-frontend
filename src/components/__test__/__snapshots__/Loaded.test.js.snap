// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loaded component Should match snapshot 1`] = `
<div>
  <withRouter(Connect(WithFormik(QuoteForm)))
    error=""
    insertAuthor="test"
    lang="test"
    quote="test"
    quoteAuthor="test"
    touched=""
  />
</div>
`;
